### PR TITLE
fix: edge case with the link protocol

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?

Fixes a bug when a project uses `"my-app": "link:."` inside its dependencies.

It does this by ensuring that we don't loop continuously on symlinks within `node_modules` that point to the original directory. I believe it's still possible to trigger infinite loops (for example with a symlink in `a/node_modules/b` pointing to `b`, and another in `b/node_modules/a` pointing to `a`, but the most problematic use case is covered).

cc @lirantal 

#### How should this be manually tested?

- Run `yarn policies set-version nightly` to fetch the latest Yarn build
- Run `yarn add my-app@link:.`
- Run `yarn add snyk`
- Run `yarn snyk monitor --file=package.json`

It shouldn't hang forever (note that `yarn snyk test` will still crash, but this one can be fixed in Yarn so that's what I'll do).

#### Any background context you want to provide?

We had to use a resolution override in our `package.json` to use my fork and avoid our CI OOMing.

#### Additional questions

Not sure I understand how the tests fixtures work so I haven't added one.

Original PR: https://github.com/snyk/resolve-deps/pull/53

### CODE REVIEW

**Comment by lili2311** [comment]
https://github.com/snyk/snyk/issues/821

File: test/fixtures/yarn-link-deps/package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "yarn-link-deps",
+  "version": "0.0.1",
+  "dependencies": {
+    "debug": "^2.2.0",
+    "my-app": "link:./linked-dep"
```
Comment (gitphill) [inline test/fixtures/yarn-link-deps/package.json:6]: test for "link:." too?

**Comment by aviadhahami** [comment]
@lili2311 @gitphill What can be done here in order to resolve this pr? :D

**Comment by dkontorovskyy** [comment]
Was brought in https://github.com/snyk/resolve-deps/pull/69

**Pull request closed without merging**